terraform/iam/iam-general.txt

```diff
@@ -0,0 +1,79 @@
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "CFGated",
+            "Effect": "Allow",
+            "Action": [
+                "*"
+            ],
+            "Resource": [
+                "arn:aws:cloudformation:<YOUR-REGION>:<IAM-ACCOUNT-NUM>:stack/<YOUR-STACK-NAME>/*",
+                "arn:aws:cloudformation:<YOUR-REGION>:<IAM-ACCOUNT-NUM>:stackset/<YOUR-STACK-NAME>:*"
+            ]
+        },
+        {
+            "Sid": "S3Gated",
+            "Effect": "Allow",
+            "Action": "s3:*",
+            "Resource": [
+                "arn:aws:s3:::<YOUR-STACK-NAME>-cf-assets/*",
+                "arn:aws:s3:::<YOUR-STACK-NAME>-cf-assets"
```

Comment:
in order to tear down a deployment, these lines need to be replaced with just `"arn:aws:s3:::<YOUR-STACK-NAME>*`